From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: darkyzhou <me@zqy.io>
Date: Sun, 14 Dec 2025 12:37:22 +0000
Subject: loong64 support

Co-authored-by: Jiajie Chen <c@jia.je>

diff --git a/scripts/build/rollup.config.mjs b/scripts/build/rollup.config.mjs
index a0ce73ccee40959c8f9d16e1e7e25733eb46ae6a..d80faddcf367a921693db7e4d9038e45d01107a7 100644
--- a/scripts/build/rollup.config.mjs
+++ b/scripts/build/rollup.config.mjs
@@ -19,7 +19,7 @@ export default commandLineArgs => ({
     sourcemap: Boolean(commandLineArgs.configSourcemaps),
   }],
   plugins: [
-    terser(),
+    // terser(),
     {
       name: 'devtools-plugin',
       resolveId(source, importer) {
